hazelcast/src/main/java/com/hazelcast/security/permission/ConnectorPermission.java

```diff
@@ -61,6 +62,21 @@ public static ConnectorPermission jdbc(@Nullable String connectionUrl, String ac
         return new ConnectorPermission(JDBC_PREFIX + (connectionUrl == null ? "" : connectionUrl), action);
     }
 
+    /**
+     * @param connectionDescription connection string or data connection name
+     * @param databaseName database name this permission is about, null if permission is granted to all databases
+     * @param collectionName collection name this permission is about, null if permission is granted to all collections
+     * @param action action this permission is about
+     */
+    public static ConnectorPermission mongo(@Nullable String connectionDescription,
+                                            @Nullable String databaseName, @Nullable String collectionName,
+                                            String action) {
+        String db = databaseName == null ? "*" : databaseName;
+        String col = collectionName == null ? "*" : collectionName;
```

Comment:
The wildcards here don't seem correct. This factory method is used for checked permissions and not the granted ones. 
If we allow `null` parameters, then I would either use the empty string or a hardcoded constant value.

Reply:
My understanding was that if user did not provide database name / collection name, then he must have permission for all dbs/collections. This is a case when user creates MongoClient by himself and we don't know if there is any specific db/col we want to protect.

Reply:
It still feels wrong to me. What about using a hardcoded value that can't be a real value?
See https://www.mongodb.com/docs/manual/reference/limits/#naming-restrictions
e.g. `$ANY$`

Reply:
https://github.com/hazelcast/hazelcast/pull/24467/commits/68f6b23867a18b2f961db1f7d054865956f7fa5f